File: src/main/java/org/example/commons/entities/filters/GroupFilter.java
package org.example.commons.entities.filters;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class GroupFilter {
    public static final GroupFilter DEFAULT = GroupFilter.builder().build();

    @Builder.Default
    private String id = "";

    @Builder.Default
    private String name = "";

    @Builder.Default
    private String description = "";
}

File: src/main/kotlin/org/example/commons/entities/filters/GroupFilter.kt
package org.example.commons.entities.filters

import org.example.commons.api.Filter
import org.example.commons.entities.Group

data class GroupFilter(
        var id: String = "",
        var name: String = "",
        var description: String = ""
) : Filter<Group> {
    companion object {
        val default: GroupFilter = GroupFilter()
    }
}